src/main/kotlin/io/zerobase/smarttracing/resources/DevicesResource.kt

```diff
@@ -1,34 +1,70 @@
 package io.zerobase.smarttracing.resources
 
-import io.zerobase.smarttracing.models.IdWrapper
+import com.fasterxml.jackson.annotation.JsonProperty
+import io.zerobase.smarttracing.GraphDao
+import io.zerobase.smarttracing.models.*
 import javax.ws.rs.*
 import javax.ws.rs.core.MediaType
 
 //region Request Models
+data class CreateDeviceRequest(@JsonProperty("fingerprint") val fingerprint: Fingerprint)
+
+data class CreateCheckInRequest(
+        @JsonProperty("scannedId") val scannedId: ScannableId,
+        @JsonProperty("type") val type: ScanType,
+        @JsonProperty("location") val location: Location?
+) {
+    enum class ScanType {
+        DEVICE_TO_DEVICE,
+        DEVICE_TO_SCANNABLE
+    }
+}
+
+data class UpdateCheckInLocationRequest(
```

Comment:
Why not just use the Location class?

Reply:
That's much smarter! Will fix.

Reply:
Realizing you probably already fixed this one, I will only fix David's comments.

Reply:
i fixed his as well, i was gonna suggest those, i have the files open and it is relatively small changes so i just fixed those first, not too sure if the enum method you used will work correctly will comment when i am doing my testing on my code to ensure it works